clearwater-infrastructure/usr/share/clearwater/bin/process-stability

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# @file process-stability
+#
+# Copyright (C) Metaswitch Networks 2017
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+# Used for monitoring the stability of a process.
+
+method=$1
+process_name=$2
+grace_period=$3
+abort_timeout=$4
+abort_flag_name="$process_name-aborting-flag"
+
+usage="Usage: check-stability {check|reset|aborted} <process name> <grace period (s)> <abort timeout>"
+
+[ $# = 4 ] || { echo $usage >&2 ; exit 2; }
+
+# Returns 1 if the process has not been running for at least the grace period,
+# or the abort flag is set (and has not timed out). Returns 2 if there is no
+# process matching the value in the pidfile. Returns 0 otherwise.
+check_stability() {
+  pidfile=/var/run/$process_name/$process_name.pid
+
+  # It's expected that there might not be a pidfile
+  pid=$( cat $pidfile 2>/dev/null)
+
+  # Determine the uptime
+  uptime=$( ps -p $pid -o etimes= 2>/dev/null) || { echo "No process matching value from pidfile: $pid " >&2 ; return 1 ; }
+
+  # Call 'timeout' on the flag - this will clear the flag if it has been set for
+  # longer than $aborting_flag_timeout
+  /usr/share/clearwater/bin/cw-flag timeout $abort_flag_name $abort_timeout
+
+  if ! /usr/share/clearwater/bin/cw-flag read $abort_flag_name; then
```

Comment:
I don't think the user should be responsible for calling timeout - this should be part of the read function.

Reply:
Sounds reasonable - I've made this change.